### ISSUE TITLE
Stress tensor shape not compatible with ASE stress tensor

### ISSUE THREAD

**Issue by kevinwhitham** [dftd4/dftd4#59]
This is an issue with the python interface to ASE. ASE and GPAW calculators return a stress tensor with shape (6,) by default. This is the Voigt style (xx, yy, zz, yz, xz, xy) (see  [ASE docs](https://wiki.fysik.dtu.dk/ase/ase/atoms.html#ase.Atoms.get_stress)) dftd4 uses a (3,3) array. This causes an error in the get_property method of the DispersionCalculator class where the ASE/GPAW calculator result is added to the dftd4 result and the array shapes do not match. A quick fix is to reshape the self_result array, but there should be more robust checking implemented.

```
if calc_result.shape == (6,) and self_result.shape == (3,3):
    self_result = np.array([self_result[0,0], self_result[1,1], self_result[2,2], self_result[1,2], self_result[0,2], self_result[0,1]])
       
return self_result + calc_result
```

**Comment by awvwgk** [comment]
So far ASE seemed able to deal with non-Voigt stress tensors automatically, but the dispersion correction situation seems to bypass this compatibility fix due to the ugly intermediate layer for merging DFT and dispersion correction result.

The stress tensor is copied over from the API in
https://github.com/dftd4/dftd4/blob/ae8031b04534714d5ff5bc9613cba8b25dd1dcd2/python/dftd4/interface.py#L285
where this should probably be fixed by a simple pack instruction, maybe even averaging the off-diagonal elements of the stress tensor when transferring to the Voigt shape.

**Comment by awvwgk** [comment]
The Python API and the ASE calculator for `dftd4` will be redone similar to the one available in `xtb` (see #66). This issue is relevant there as well.
